pylint/checkers/strings.py

```diff
@@ -238,6 +245,25 @@ def get_access_path(key, parts):
             path.append("[{!r}]".format(specifier))
     return str(key) + "".join(path)
 
+def arg_matches_format_type(arg_type, format_type):
+#    print('arg_type = %s' % arg_type)
+    if arg_type in (None, astroid.Uninferable):
+        return True
+    if format_type in "sr":
+        # All types can be printed with %s and %r
+        return True
+    if isinstance(arg_type, astroid.Instance):
+        arg_type = arg_type.pytype()
+#        print('pytype = %s' % arg_type)
+        if arg_type == BUILTINS + ".str":
```

Comment:
Let's put these into some constants, and use those instead.

Reply:
Done.